src/commands/general/init.ts

```diff
@@ -17,19 +17,38 @@ function getRequirementsErrorMessage({git, docker}: Record<string, boolean>): st
   if (!docker) {
     return "Docker is not installed. Please install Docker and try again.\n";
   }
+
   return "";
 }
 
+function getVersionErrorMessage({docker, node}: Record<string, string>): string {
+  let message = "";
+
+  if (docker) {
+    message += `Docker version ${docker} or higher is required. Please update Docker and try again.\n`;
+  }
+
+  if (node) {
+    message += `Node version ${node} or higher is required. Please update Node and try again.\n`;
+  }
+
+  return message;
+}
+
 export async function initAction(options: InitActionOptions, simulatorService: ISimulatorService) {
-  // Check if git and docker are installed
+  // Check if requirements are installed and if the versions are correct
   try {
-    const {git, docker} = await simulatorService.checkRequirements();
-    const errorMessage = getRequirementsErrorMessage({git, docker});
-    if (errorMessage) {
+    const requirementsInstalled = await simulatorService.checkInstallRequirements();
+    const requirementErrorMessage = getRequirementsErrorMessage(requirementsInstalled);
+
+    const missingVersions = await simulatorService.checkVersionRequirements();
+    const versionErrorMessage = getVersionErrorMessage(missingVersions);
+
+    if (requirementErrorMessage || versionErrorMessage) {
```

Comment:
@denishacquin, the separation is better, but let's execute them in different steps: first, the installRequirements and then the versionRequirements. If you want to make both checks in the same step, then the message for the user should be only one. Otherwise, they will get one message, fix it, and then get the other message. This adds complexity and affects readability.

Reply:
I split them in two separate steps like suggested.